From 92ece07809b49131f78bb6fd47ee0130d0827f3e Mon Sep 17 00:00:00 2001
From: AlexTMjugador <AlexTMjugador@users.noreply.github.com>
Date: Mon, 2 Sep 2019 18:43:04 +0200
Subject: [PATCH] package/lvm2: further fix musl compilation

For some reason, LVM2 developers always assumed that stdin, stdout and
stderr are read/write variables, even though there is freopen to
reassign them. Fix that to allow musl to complete the build.

Signed-off-by: AlexTMjugador <AlexTMjugador@users.noreply.github.com>
---
 tools/lvmcmdline.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/lvmcmdline.c b/tools/lvmcmdline.c
index 7d29b6fab..f3c015bc8 100644
--- a/tools/lvmcmdline.c
+++ b/tools/lvmcmdline.c
@@ -3139,7 +3139,7 @@ static int _check_standard_fds(void)
 	int err = is_valid_fd(STDERR_FILENO);
 
 	if (!is_valid_fd(STDIN_FILENO) &&
-	    !(stdin = fopen(_PATH_DEVNULL, "r"))) {
+	    !freopen(_PATH_DEVNULL, "r", stdin)) {
 		if (err)
 			perror("stdin stream open");
 		else
@@ -3149,7 +3149,7 @@ static int _check_standard_fds(void)
 	}
 
 	if (!is_valid_fd(STDOUT_FILENO) &&
-	    !(stdout = fopen(_PATH_DEVNULL, "w"))) {
+	    !freopen(_PATH_DEVNULL, "w", stdout)) {
 		if (err)
 			perror("stdout stream open");
 		/* else no stdout */
@@ -3157,7 +3157,7 @@ static int _check_standard_fds(void)
 	}
 
 	if (!is_valid_fd(STDERR_FILENO) &&
-	    !(stderr = fopen(_PATH_DEVNULL, "w"))) {
+	    !freopen(_PATH_DEVNULL, "w", stderr)) {
 		printf("stderr stream open: %s\n",
 		       strerror(errno));
 		return 0;
-- 
2.20.1

